control-plane/connect-inject/lifecycle/lifecycle_configuration_test.go

```diff
@@ -327,6 +393,59 @@ func TestLifecycleConfig_GracefulShutdownPath(t *testing.T) {
 	}
 }
 
+func TestLifecycleConfig_GracefulStartupPath(t *testing.T) {
+	cases := []struct {
+		Name            string
+		Pod             func(*corev1.Pod) *corev1.Pod
+		LifecycleConfig Config
+		Expected        string
+		Err             string
+	}{
+		{
+			Name: "Sidecar proxy lifecycle graceful startup path defaults to /graceful_startup",
+			Pod: func(pod *corev1.Pod) *corev1.Pod {
+				return pod
+			},
+			Expected: "/graceful_startup",
+			Err:      "",
+		},
+		{
+			Name: "Sidecar proxy lifecycle graceful startup path set via meshWebhook",
+			Pod: func(pod *corev1.Pod) *corev1.Pod {
+				return pod
+			},
+			LifecycleConfig: Config{
+				DefaultGracefulStartupPath: "/quit",
+			},
+			Expected: "/quit",
+			Err:      "",
+		},
+		{
+			Name: "Sidecar proxy lifecycle graceful port set via annotation",
+			Pod: func(pod *corev1.Pod) *corev1.Pod {
+				pod.Annotations[constants.AnnotationSidecarProxyLifecycleGracefulStartupPath] = "/custom-startup-path"
+				return pod
+			},
+			LifecycleConfig: Config{
+				DefaultGracefulStartupPath: "/quit",
+			},
```

Comment:
Sorry, missed these on the first pass.

Not a big deal, but we could also fix the test case name above for shutdown here: https://github.com/hashicorp/consul-k8s/pull/3878/files#diff-063b1fb15d1bdfce69ac658b5824bff783caacdc13d979daea5bd0dfbfe87e89L305

```suggestion
			LifecycleConfig: Config{
				DefaultGracefulStartupPath: "/start",
			},
			Expected: "/start",
			Err:      "",
		},
		{
			Name: "Sidecar proxy lifecycle graceful startup path set via annotation",
			Pod: func(pod *corev1.Pod) *corev1.Pod {
				pod.Annotations[constants.AnnotationSidecarProxyLifecycleGracefulStartupPath] = "/custom-startup-path"
				return pod
			},
			LifecycleConfig: Config{
				DefaultGracefulStartupPath: "/start",
			},
```